contracts/core/src/octane/Staking.sol

```diff
@@ -85,39 +110,50 @@ contract Staking is OwnableUpgradeable {
         emit Delegate(msg.sender, validator, msg.value);
     }
 
+    /**
+     * @notice Returns the list of allowed validator addresses
+     */
+    function allowlist() external view returns (address[] memory) {
+        return _allowlist;
+    }
+
     //////////////////////////////////////////////////////////////////////////////
     //                                  Admin                                   //
     //////////////////////////////////////////////////////////////////////////////
 
     /**
-     * @notice Enable the validator allowlist
+     * @notice Add validators to the allowlist
+     * @param enabled True to enable the allowlist, false to disable
      */
-    function enableAllowlist() external onlyOwner {
-        isAllowlistEnabled = true;
-    }
+    function setAllowlistEnabled(bool enabled) external onlyOwner {
+        require(isAllowlistEnabled != enabled, "Staking: already set");
 
-    /**
-     * @notice Disable the validator allowlist
-     */
-    function disableAllowlist() external onlyOwner {
-        isAllowlistEnabled = false;
-    }
+        isAllowlistEnabled = enabled;
 
-    /**
-     * @notice Add validators to allow list
-     */
-    function allowValidators(address[] calldata validators) external onlyOwner {
-        for (uint256 i = 0; i < validators.length; i++) {
-            isAllowedValidator[validators[i]] = true;
+        if (enabled) {
+            emit AllowlistEnabled();
+        } else {
+            emit AllowlistDisabled();
         }
     }
 
     /**
-     * @notice Remove validators from allow list
+     * @notice Add validators to the allowlist
+     * @param validators The list of validators to add
      */
-    function disallowValidators(address[] calldata validators) external onlyOwner {
+    function setAllowlist(address[] calldata validators) external onlyOwner {
+        // reset current
+        for (uint256 i = 0; i < _allowlist.length; i++) {
+            isAllowedValidator[_allowlist[i]] = false;
+        }
+
+        _allowlist = validators;
+
+        // set new
         for (uint256 i = 0; i < validators.length; i++) {
-            isAllowedValidator[validators[i]] = false;
+            isAllowedValidator[validators[i]] = true;
         }
+
+        emit AllowlistSet(validators);
     }
```

Comment:
Changes here amount to:

- replace `disableAllowlist / enableAllowlist` with `setAllowlistEnabled`
- replace  `allowValidators / disallowValidators` with `setAllowlist`